test/helpers/kubectl.go

```diff
@@ -3806,7 +3806,7 @@ func (kub *Kubectl) ciliumServicePreFlightCheck() error {
 // commands are run on all pods matching selector
 func (kub *Kubectl) reportMapContext(ctx context.Context, path string, reportCmds map[string]string, ns, selector string) {
 	for cmd, logfile := range reportCmds {
-		results, err := kub.ExecInPods(ctx, ns, selector, cmd, true, ExecOptions{SkipLog: true})
+		results, err := kub.ExecInPods(ctx, ns, selector, cmd, false, ExecOptions{SkipLog: true})
```

Comment:
I'm not sure if that's the correct way to fix it, but in any case, it doesn't do what the commit message describes. As far as I can see, this boolean instructs `ExecInPods` to wait for the pods to be ready before running the log collection commands. Setting the boolean to true or false won't affect parallelization.

On whether this is the correct fix: We likely do need to wait for the pods to be ready, at least the first time we call `ExecInPods()`. @nebril can probably confirm.

Reply:
Hmm, my understanding is that if `wait` is set to `false`, then `ExecInPods` would NOT wait for pods to be ready before executing the cmds, and when the pods are indeed NOT ready, would record cmd failure. Am I missing something? https://github.com/cilium/cilium/blob/master/test/helpers/kubectl.go#L3154

Reply:
Ah, you are right, it's not about parallelization, just skips the wait.

Reply:
> then `ExecInPods` would NOT wait for pods to be ready before executing the cmds, and when the pods are indeed NOT ready, would record cmd failure.

No, that's my understanding as well. But in some cases we might want to wait for the pods because we just deployed them (e.g., log gatherers).

Reply:
Maybe a better solution is to call `ExecPodCmdBackground` instead so that these cmds ARE parallelized, and the total wait time should be roughly `1 * helper.Timeout` instead of `len(cmds) * helper.Timeout`?

Reply:
Yep that sounds worth a try, but it's more likely to have other consequences we're missing. Did you check the executed commands to ensure we don't risk overwriting the result of one command with another if we do that? I would expect we write to different files, but maybe worth double checking.

Reply:
Yea `reportMapContext` handles the actual fs operations and it is using different logfile for different cmd. I'll update the commit to try out this solution.

Reply:
Let's iterate on the new solution, closing this one since this was probably not the right fix.